### PR TITLE
Fix crash during downloading a new release

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Fix crash during downloading a new release with a specific Gradle configuration.

**Repo steps:**
- Create empty app.
- Add these lines to the app gradle:
```
release {
    minifyEnabled true
    proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
}
```
- Run the app and press on download button on the in-app update dialog.

**Expected behavior:**
Release successfully downloaded.

**Actual behavior:**
App crashes.

## Related PRs or issues

[AB#86011](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/86011)

### CODE REVIEW

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java [hunk 1]
```diff
@@ -57,6 +57,24 @@ protected Void doInBackground(Void... params) {
              * IllegalArgumentException: Unknown URL content://downloads/my_download is thrown.
              */
             mDownloader.onDownloadError(new IllegalStateException("Failed to start download: Download Manager is disabled.", e));
+        } catch (Throwable throwable) {
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java:60]: Can it be more specific?

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java [hunk 2]
```diff
@@ -57,6 +57,24 @@ protected Void doInBackground(Void... params) {
...
+             * }
+             *
+             * Possible error:
+             * Caused by: java.lang.VerifyError: Verifier rejected class b.c.a.n.j.c.c: java.lang.Object
+             * b.c.a.n.j.c.c.doInBackground(java.lang.Object[]) failed to verify:
+             * java.lang.Object b.c.a.n.j.c.c.doInBackground(java.lang.Object[]): [0x6E]
+             * expected to be within a catch-all for an instruction where a monitor is held
+             * (declaration of 'b.c.a.n.j.c.c' appears in base.apk).
+             */
+            throw throwable;
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java:77]: We just re-throwing it, could you please explain how exactly it fixes the crash?

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java (hunk 1 above)
Comment (AnastasiaKubova) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java:60]: nope

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java (hunk 2 above)
Comment (AnastasiaKubova) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java:77]: I explained it in the [comments](https://github.com/microsoft/appcenter-sdk-android/pull/1525/files#diff-26adf120c5910d7c6e54457267877a2e6637406f6a9d2f534367e86c5a13e224R74).

**Review by MatkovIvan — CHANGES_REQUESTED** [review]
Looks like invalid fix that affects [internal ART validation logic](https://android.googlesource.com/platform/art/+/master/runtime/verifier/method_verifier.cc#3623).

As described in the issue, let's modify regular proguard rules instead

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**